File: Domain/Entities/Item.cs
using Domain.Enum;
using Domain.Interfaces;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities
{
	public class Item : WithId, IWithSrc
	{
		public string Title { get; set; }
		public string Description { get; set; }

		public Tier Tier { get; set; }

		public string Src { get; set; }

		[ForeignKey(nameof(CategoryId))]
		public Category Category { get; set; }
		public Guid CategoryId { get; set; }
	}
}

File: WebApi/ClientApp/src/objects/Item.ts
import { Category } from "./Category";
import { Tier } from "./enums";

export interface Item {
	id: string;
	title: string;
	description: string;
	src: string;
	tier: Tier;
	category: Category | undefined;
	categoryId: string;
}

export const tierColors = {
	[Tier.S]: '#ffa333',
	[Tier.A]: '#ad33ff',
	[Tier.B]: '#4c57f5',
	[Tier.C]: '#26eb54',
	[Tier.D]: '#a1a1a1',
	[Tier.None]: '#ffff',
};

export const tierNames = {
	[Tier.S]: 'S',
	[Tier.A]: 'A',
	[Tier.B]: 'B',
	[Tier.C]: 'C',
	[Tier.D]: 'D',
	[Tier.None]: 'None',
};